In [1]:
#importing libraries 
import pandas as pd
import numpy as np
import os

In [3]:
#dropping eval_set column from orders.csv
path = r'C:\Users\lenovo\09-2024 CareerFoundry'
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'orders.csv'))
df_ords = df_ords.drop(columns = ['eval_set'])

In [4]:
#renaming 'order_dow' column
df_ords.rename(columns = {'order_dow' : 'orders_day_of_week'}, inplace = True)

In [5]:
df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [6]:
#changing the column data type of 'order_id' to str
df_ords['order_id'] = df_ords['order_id'].astype('str')

In [7]:
#importing data set departments.csv
path = r'C:\Users\lenovo\09-2024 CareerFoundry'
df_dep = pd.read_csv(os.path.join(path, '02 Data', 'departments.csv'))
df_dep.columns

Index(['department_id', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10',
       '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21'],
      dtype='object')

In [10]:
#Transposing df_dep
df_dep_t = df_dep.T

In [11]:
df_dep_t.reset_index()

index                0
0   department_id       department
1               1           frozen
2               2            other
3               3           bakery
4               4          produce
5               5          alcohol
6               6    international
7               7        beverages
8               8             pets
9               9  dry goods pasta
10             10             bulk
11             11    personal care
12             12     meat seafood
13             13           pantry
14             14        breakfast
15             15     canned goods
16             16       dairy eggs
17             17        household
18             18           babies
19             19           snacks
20             20             deli
21             21          missing

In [12]:
# Take the first row of df_dep_t for the header
new_header = df_dep_t.iloc[0]

In [13]:
new_header

0    department
Name: department_id, dtype: object

In [15]:
df_dep_t_new = df_dep_t[1:]
df_dep_t_new

0
1            frozen
2             other
3            bakery
4           produce
5           alcohol
6     international
7         beverages
8              pets
9   dry goods pasta
10             bulk
11    personal care
12     meat seafood
13           pantry
14        breakfast
15     canned goods
16       dairy eggs
17        household
18           babies
19           snacks
20             deli
21          missing

In [16]:
df_dep_t_new.columns = new_header # set the header row as the df header
df_dep_t_new

department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

Data Dictionary

In [18]:
data_dict = df_dep_t_new.to_dict('index')
data_dict

{'1': {'department': 'frozen'},
 '2': {'department': 'other'},
 '3': {'department': 'bakery'},
 '4': {'department': 'produce'},
 '5': {'department': 'alcohol'},
 '6': {'department': 'international'},
 '7': {'department': 'beverages'},
 '8': {'department': 'pets'},
 '9': {'department': 'dry goods pasta'},
 '10': {'department': 'bulk'},
 '11': {'department': 'personal care'},
 '12': {'department': 'meat seafood'},
 '13': {'department': 'pantry'},
 '14': {'department': 'breakfast'},
 '15': {'department': 'canned goods'},
 '16': {'department': 'dairy eggs'},
 '17': {'department': 'household'},
 '18': {'department': 'babies'},
 '19': {'department': 'snacks'},
 '20': {'department': 'deli'},
 '21': {'department': 'missing'}}

In [20]:
#Import the “products.csv” file into Jupyter as df_prods
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'products.csv'))
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

Subsetting

In [21]:
df_snacks =  df_prods[df_prods['department_id']==19]
df_snacks.head()

product_id                                 product_name  aisle_id  \
0            1                   Chocolate Sandwich Cookies        61   
15          16                Mint Chocolate Flavored Syrup       103   
24          25      Salted Caramel Lean Protein & Fiber Bar         3   
31          32                Nacho Cheese White Bean Chips       107   
40          41  Organic Sourdough Einkorn Crackers Rosemary        78   

    department_id  prices  
0              19     5.8  
15             19     5.2  
24             19     1.9  
31             19     4.9  
40             19     6.5

In [22]:
df_snacks_2 = df_prods.loc[df_prods['department_id'] == 19]
df_snacks_3 = df_prods.loc[df_prods['department_id'].isin([19])]

#TASK WRANGLING

Q. Find another identifier variable in the df_ords dataframe that doesn’t need to be included in your analysis as a numeric variable and change it to a suitable format


In [23]:
df_ords['user_id'] = df_ords['user_id'].astype('str')

Q. Look for a variable in your df_ords dataframe with an unintuitive name and change its name without overwriting the dataframe

In [25]:
df_ords.rename(columns = {'order_dow' : 'orders_day_of_week'}, inplace = False) #since all other column names are intuitive, only 'order_dow' needs to be renamed

order_id user_id  order_number  orders_day_of_week  order_hour_of_day  \
0        2539329       1             1                   2                  8   
1        2398795       1             2                   3                  7   
2         473747       1             3                   3                 12   
3        2254736       1             4                   4                  7   
4         431534       1             5                   4                 15   
...          ...     ...           ...                 ...                ...   
3421078  2266710  206209            10                   5                 18   
3421079  1854736  206209            11                   4                 10   
3421080   626363  206209            12                   1                 12   
3421081  2977660  206209            13                   1                 12   
3421082   272231  206209            14                   6                 14   

         days_since_prior_order  
0                           NaN  
1                          15.0  
2                          21.0  
3                          29.0  
4                          28.0  
...                         ...  
3421078                    29.0  
3421079                    30.0  
3421080                    18.0  
3421081                     7.0  
3421082                    30.0  

[3421083 rows x 6 columns]

Q. Your client wants to know what the busiest hour is for placing orders. Find the frequency of the corresponding variable and share your findings

In [27]:
df_ords['order_hour_of_day'].value_counts(dropna = True)

10    288418
11    284728
15    283639
14    283042
13    277999
12    272841
16    272553
9     257812
17    228795
18    182912
8     178201
19    140569
20    104292
7      91868
21     78109
22     61468
23     40043
6      30529
0      22758
1      12398
5       9569
2       7539
4       5527
3       5474
Name: order_hour_of_day, dtype: int64

The busiest hour of the day is 10 AM

Q. Determine the meaning behind a value of 4 in the "department_id" column within the df_prods dataframe using a data dictionary.

In [29]:
print(data_dict.get('4'))

{'department': 'produce'}


Department ID 4 refers to the 'produce' department

Q. The sales team in your client’s organization wants to know more about breakfast item sales. Create a subset containing only the required information.

In [30]:
data_dict

{'1': {'department': 'frozen'},
 '2': {'department': 'other'},
 '3': {'department': 'bakery'},
 '4': {'department': 'produce'},
 '5': {'department': 'alcohol'},
 '6': {'department': 'international'},
 '7': {'department': 'beverages'},
 '8': {'department': 'pets'},
 '9': {'department': 'dry goods pasta'},
 '10': {'department': 'bulk'},
 '11': {'department': 'personal care'},
 '12': {'department': 'meat seafood'},
 '13': {'department': 'pantry'},
 '14': {'department': 'breakfast'},
 '15': {'department': 'canned goods'},
 '16': {'department': 'dairy eggs'},
 '17': {'department': 'household'},
 '18': {'department': 'babies'},
 '19': {'department': 'snacks'},
 '20': {'department': 'deli'},
 '21': {'department': 'missing'}}

In [32]:
#'breakfast' dataframe
df_prods_bf = df_prods[df_prods['department_id']==14]

Q.Find all observations from the entire dataframe that include items from the following departments: alcohol, deli, beverages, and meat/seafood.

In [33]:
#'dinner party' dataframe
df_prods_dinner = df_prods.loc[df_prods['department_id'].isin([5,7,12,20])]

Q. How many rows does the last dataframe you created have?

In [34]:
df_prods_dinner.shape

(7650, 5)

The 'dinner party' dataframe has 7650 rows

Q. Extract all the information you can about user with user_id = 1

In [41]:
df_ords_user1 = df_ords.loc[df_ords['user_id']=='1']

Q. Stats about this user

In [42]:
df_ords_user1.describe()

order_number  orders_day_of_week  order_hour_of_day  \
count     11.000000           11.000000          11.000000   
mean       6.000000            2.636364          10.090909   
std        3.316625            1.286291           3.477198   
min        1.000000            1.000000           7.000000   
25%        3.500000            1.500000           7.500000   
50%        6.000000            3.000000           8.000000   
75%        8.500000            4.000000          13.000000   
max       11.000000            4.000000          16.000000   

       days_since_prior_order  
count               10.000000  
mean                19.000000  
std                  9.030811  
min                  0.000000  
25%                 14.250000  
50%                 19.500000  
75%                 26.250000  
max                 30.000000

Q. Export your df_ords dataframe as “orders_wrangled.csv” in your “Prepared Data” folder

In [44]:
df_ords.to_csv(os.path.join((os.path.join(path, '02 Data','Prepared Data', 'orders_wrangled.csv'))))

Q. Export the df_dep_t_new dataframe as “departments_wrangled.csv” in your “Prepared Data” folder so that you have a “.csv” file of your departments data in the correct format

In [ ]:
df_dep_t_new.to_csv(os.path.join((os.path.join(path, '02 Data','Prepared Data', 'departments_wrangled.csv'))))